### PR TITLE
Feat/relay service

### PR DESCRIPTION
**Pull request by maycon-mello**

### CODE REVIEW

File: packages/core/lib/modules/wallet.js [hunk 1]
```diff
@@ -274,7 +272,7 @@ class Wallet {
       throw new Error(
         'Wallet module timed out. Make sure the wallet is loaded, or you are not using multiple instances',
       );
-    }, 6000);
+    }, 30000);
```
Comment (mike-parkhill) [inline packages/core/lib/modules/wallet.js:275]: that's a pretty big change - what's the reasoning for going that big?

File: packages/core/lib/services/relay-service/configs.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import assert from 'assert';
+import {isNumberValid} from '../../core/validation';
+
+export const validation = {
+  sendMessage({keyPairDoc, recipientDid, message}: SendMessageParams) {
+    assert(!!keyPairDoc, 'keyPairDoc is required');
+    assert(!!recipientDid, 'recipientDid is required');
+    assert(typeof message === 'string', 'invalid message');
```
Comment (mike-parkhill) [inline packages/core/lib/services/relay-service/configs.js:8]: would there be value to accepting an object and serializing it? Or are you deliberately leaving serialization to the caller?

File: packages/relay-service/lib/index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import assert from 'assert';
+import axios from 'axios';
+import {generatePayload, toBase64} from './payloads';
+
+let serviceURL = 'https://relay.dock.io';
```
Comment (mike-parkhill) [inline packages/relay-service/lib/index.js:5]: can we read this from a config? Thinking at some point we may want to be able to change it (test env, white label clients running their own, etc.)

File: packages/relay-service/lib/payloads.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import VerifiableCredential from '@docknetwork/sdk/verifiable-credential';
+import {cryptoWaitReady} from '@polkadot/util-crypto';
+import {getKeypairFromDoc} from '@docknetwork/universal-wallet/methods/keypairs';
+import {getSuiteFromKeyDoc} from '@docknetwork/sdk/utils/vc/helpers';
+
+// 1 year
+const DEFAULT_EXPIRATION = 86400 * 1000 * 365;
+
+export async function generatePayload(keyPairDoc, subject) {
```
Comment (mike-parkhill) [inline packages/relay-service/lib/payloads.js:9]: should probably rename this to indicate what kind of payload it generates

File: packages/relay-service/lib/payloads.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      RelayAuthCredential: 'dk:RelayAuthCredential',
+      limit: 'dk:limit',
+      to: 'dk:to',
+      msg: 'dk:msg',
+    },
+  ]);
+
+  cred.setIssuanceDate(new Date().toISOString());
+  cred.setExpirationDate(
+    new Date(Date.now() + DEFAULT_EXPIRATION).toISOString(),
```
Comment (mike-parkhill) [inline packages/relay-service/lib/payloads.js:26]: would it make sense to have to renew this more frequently?

Comment (mike-parkhill) [reply packages/relay-service/lib/payloads.js:26]: in a similar vein, have we given any thought to de-authorizing the connection in general?

Comment (maycon-mello) [reply packages/relay-service/lib/payloads.js:26]: I can handle it during this week

File: packages/relay-service/lib/index.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import assert from 'assert';
+import axios from 'axios';
+import {generateSignedPayload, toBase64} from './payloads';
+
+let serviceURL = process.env.RELAY_SERVICE_URL || 'https://relay.dock.io';
+
+const sendMessage = async ({keyPairDoc, recipientDid, message}) => {
+  assert(!!keyPairDoc, 'senderDid is required');
```
Comment (mike-parkhill) [inline packages/relay-service/lib/index.js:8]: Message doesn't match the parameter name

**Review by mike-parkhill — COMMENTED** [review]
One little nitpick

**Comment by maycon-mello** [comment]
can you please check again @mike-parkhill

**Review by mike-parkhill — APPROVED** [review]

**Pull request merged**